File: src/main/java/com/technews/aggregate/posts/constant/SpringBlogsSubject.java
package com.technews.aggregate.posts.constant;

import lombok.AllArgsConstructor;

@AllArgsConstructor
public enum SpringBlogsSubject {
    ENGINEERING("engineering"),
    RELEASES("releases"),
    NEWS("news"),
    ;

    private String value;

    public String value() {
        return value;
    }
}

File: src/main/kotlin/com/technews/aggregate/posts/constant/SpringBlogsSubject.kt
package com.technews.aggregate.posts.constant

enum class SpringBlogsSubject(val value: String) {
    ENGINEERING("Engineering"),
    RELEASES("Releases"),
    NEWS("News"),
    ;

    companion object {
        fun from(value: String): SpringBlogsSubject {
            return entries.firstOrNull { it.value == value } ?: ENGINEERING
        }
    }
}
